base/irrationals.jl

```diff
@@ -56,6 +56,17 @@ float(::Type{<:AbstractIrrational}) = Float64
 ==(::Irrational{s}, ::Irrational{s}) where {s} = true
 ==(::AbstractIrrational, ::AbstractIrrational) = false
 
+<(::Irrational{s}, ::Irrational{s}) where {s} = false
+function <(x::AbstractIrrational, y::AbstractIrrational)
+    # If two different Irrationals round to the same Float64, then explicit
+    # < methods are needed for those inputs, as the below assertation will fail.
+    @assert Float64(x) != Float64(y)
```

Comment:
This should probably throw some specific error.

Reply:
Probably a good idea. Something like
```
Float64(x) != Float64(y) || error("$x and $y have identical Float64 representations")
```

Reply:
Or a `MethodError`. The solution is to add a specific method, after all.

Reply:
I've added a commit, changing the code to throw a `MethodError`.